broker.go

```diff
@@ -156,6 +157,17 @@ func NewBroker(addr string) *Broker {
 	return &Broker{id: -1, addr: addr}
 }
 
+func (b *Broker) getSockError() error {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+
+	if b.connTCP == nil {
+		return nil
+	}
+
+	return getTCPConnSockError(b.connTCP)
```

Comment:
I continue to have concerns about using a different variable to store the same value that would already be available via `b.conn.(*net.TCPConn)`.

There is no need to coördinate handling both the `conn` and `connTCP` variables to ensure that they remain consistent with each other. In context of the [Swiss-Cheese Model](https://en.wikipedia.org/wiki/Swiss_cheese_model) of accident causation, forcing developers to manually ensure the consistency of two variables when it can be driven by just the one variable would open up a significant hole in one of the layers of swiss cheese, making accidents during refactoring more likely.

```suggestion
	connTCP, _ := b.conn.(*net.TCPConn)
	if connTCP == nil {
		return nil
	}

	return getTCPConnSockError(connTCP)
```